### ISSUE TITLE
Remove `date_default_timezone_set` rule from outputting on VIP Go environments.

### ISSUE THREAD

**Issue by mdbitz** [Automattic/VIP-Coding-Standards#426]
## What problem would the enhancement address for VIP?

Reduce False Positive Warnings to Client Developers and VIP colleagues

## Describe the solution you'd like

Remove bellow code sniff from triggering on VIP Go powered sites.

```
🚫 Error: Using date_default_timezone_set() and similar isn't allowed, instead use WP internal timezone support (WordPress.WP.TimezoneChange.timezone_change_date_default_timezone_set).
```

## What code should be reported as a violation?

Nothing this rule is only required for WP.com and does not apply to VIP Go

**Comment by GaryJones** [comment]
This function is marked as a violation in [the `WordPress-Extra` ruleset](https://github.com/WordPress/WordPress-Coding-Standards/blob/3b2f88125455c27836dfdab910db25b0e76261d9/WordPress-Extra/ruleset.xml#L148). 

Does it make sense to allow the use of this behaviour, when using WP internal timezone support (however broken it might be) is expected?

**Comment by rebeccahum** [comment]
The reason why we don't allow this on WordPress.com because it breaks things afaik: https://lobby.vip.wordpress.com/wordpress-com-documentation/use-current_time-not-date_default_timezone_set/.  Also seeing docs for it for VIP Go: https://wpvip.com/documentation/vip-go/use-current_time-not-date_default_timezone_set/.

If we are removing the sniff on VIP Go, we should update our docs.

**Comment by GaryJones** [comment]
Our docs say (my emphasis):

> "that’s what **WordPress** expects it to be. Several features are dependent on this, and will break if you adjust the timezone."

That seems fairly clear cut, even if not comprehensive, that usage of `date_default_timezone_set()` may do Bad Things within WordPress, regardless of platform or configuration.

As such, I'm going to close this Issue and we'll keep reporting usage of the function via its continued [inclusion in the `WordPressVIPMinimum` ruleset](https://github.com/Automattic/VIP-Coding-Standards/blob/adf7bc93a1c77baa363a3b2d2bb9bab54cf77306/WordPressVIPMinimum/ruleset.xml#L29).
